src/modules/accounting/accounting.py

```diff
@@ -167,8 +167,7 @@ def _get_finalization_shares_rate(self, blockstamp: BlockStamp) -> int:
         simulation = self.get_rebase_after_report(blockstamp)
         return simulation.post_total_pooled_ether * SHARE_RATE_PRECISION_E27 // simulation.post_total_shares
 
-    @lru_cache(maxsize=1)
-    def get_rebase_after_report(self, blockstamp: BlockStamp) -> LidoReportRebase:
+    def get_rebase_after_report(self, blockstamp: BlockStamp, cl_only=False) -> LidoReportRebase:
```

Comment:
I would suggest do not have a flag that changes the method behaviour. It could be refactored to smth like this:

```python
def get_full_rebase_after_report(blockstamp):
  wc_balance = self._get_withdrawal_balance(blockstamp)
  el_balance = self._get_el_vault_balance(blockstamp)
  return simulate_oracle_report(wc_balance, el_balance)

def get_cl_rebase_after_report(blockstamp):
  wc_balance = self._get_withdrawal_balance(blockstamp)
  return simulate_oracle_report(wc_balance, el_balance=0)

def simulate_oracle_report(wc_balance, el_balance):
  # ...
  # calls self.w3.lido_contracts.lido.functions.handleOracleReport
```